lib/spack/docs/conf.py

```diff
@@ -110,13 +114,13 @@
 for line in fileinput.input('spack.rst', inplace=1):
     if handling_spack:
         if not line.startswith('    :noindex:'):
-            print '    :noindex: %s' % ' '.join(spack.__all__)
+            print('    :noindex: %s' % ' '.join(spack.__all__))
         handling_spack = False
 
     if line.startswith('.. automodule::'):
         handling_spack = (line == '.. automodule:: spack\n')
 
-    print line,
+    sys.stdout.write(line)
```

Comment:
There's no way to do print-with-comma with the new print function.  Using write() omits the newline.